kolena/_experimental/search/embeddings.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import dataclasses
+import json
+from typing import List
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+from dacite import from_dict
+
+from kolena._api.v1.generic import Search as API
+from kolena._experimental.search._internal.datatypes import LocatorEmbeddingsDataFrameSchema
+from kolena._utils import krequests
+from kolena._utils import log
+from kolena._utils.batched_load import init_upload
+from kolena._utils.batched_load import upload_data_frame
+from kolena._utils.consts import BatchSize
+from kolena._utils.dataframes.validators import validate_df_schema
+
+
+def upload_embeddings(embeddings: List[Tuple[str, np.typing.ArrayLike]]) -> None:
```

Comment:
The more I look at this, the more I think we should include `model_locator` as part of the data bundle (instead of inferring from the galleon table)

Reply:
Do we control the model?

Reply:
Would users call this in the same script that they call `extract_embeddings` from?

Reply:
> Do we control the model, i.e. can we handle user-provided data here?

We currently bake the model into the bundle that we provide to our customers.

> Would users call this in the same script that they call extract_embeddings from?

Hmm we could update the `extract_embeddings` script to include an `upload_embeddings(images: List[PIL.Image.Image])` entrypoint that automatically handles upload. Taking a look.

Reply:
Okay, updated this approach -- including `key` to act as a user friendly `model_locator` (since it actually just acts as a key).

Proposing [this](https://github.com/kolenaIO/kolena-studio-client-api/pull/13/files#diff-34b4a4dec40fd50e2de6307f12d8c3a2578fdc9deef4290ccdb66c040bdb5dbaR21) change for how users end up calling into this endpoint.